### PR TITLE
[Build] Use swiftc tool for release mode

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by ddunbar** [comment]
It really bugs me that we have very different behavior for `debug` vs `release`, can we just resolve that?

**Comment by aciidb0mb3r** [comment]
What do you mean? Use swift tool instead of shell for release builds?

**Comment by ddunbar** [comment]
Yeah, they have pretty different behavior (e.g., release builds aren't properly handling dependencies because of this).

**Comment by aciidb0mb3r** [comment]
done but It looks like to use `-whole-module-optimization` need to hack a bit on llbuild and generate different `-output-file-map` where `""` contains paths to `.o` and `.d`, I'll try to get that in soon.

**Comment by aciidb0mb3r** [comment]
This PR is now dependent on merge of https://github.com/apple/swift-llbuild/pull/28

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@mxcl @ddunbar please review

File: Sources/Build/Command.compile(SwiftModule).swift [hunk 1]
```diff
@@ -20,36 +20,23 @@ extension Command {
...
-            let tool = ShellTool(
-                description: "Compile \(module.name)",
-                inputs: inputs,
-                outputs: [productPath, module.targetName],
-                args: [SWIFT_EXEC, "-o", productPath] + args + module.sources.paths + otherArgs)
+        let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: args + otherArgs, executable: SWIFT_EXEC, conf: conf)
 
-            return (cmd(tool), [])
-        }
+        //FIXME these should be inferred as implicit inputs by llbuild
```
Comment (ddunbar) [inline Sources/Build/Command.compile(SwiftModule).swift]: OT, but I believe this is fixed and can die.

File: Sources/Build/Command.compile(SwiftModule).swift [hunk 2]
```diff
@@ -20,36 +20,23 @@ extension Command {
             return Command(node: module.targetName, tool: tool)
         }
 
+        var args = ["-j8", "-D", "SWIFT_PACKAGE"]
```
Comment (ddunbar) [inline Sources/Build/Command.compile(SwiftModule).swift]: Let's use the same `numThreads` constant for this, so it isn't hard coded twice.

**Comment by ddunbar** [comment]
LGTM, one request which can be cleaned up in a follow on.

**Comment by aciidb0mb3r** [comment]
followup here:  https://github.com/apple/swift-package-manager/pull/319

**Pull request merged**